### PR TITLE
Correct URL reference to downloads

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

**Review by lrknox — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -103,14 +103,18 @@ in which a feature is introduced may change.
...
+
    https://github.com/HDFGroup/hdf5/releases/tag/snapshot
 
 Source packages for current and previous releases are located at:
-    
-   https://support.hdfgroup.org/downloads/hdf5
+
+   hdf5 1.14 releases:
+   https://support.hdfgroup.org/releases/hdf5/downloads
+
```
Comment (hyoklee) [inline README.md:113]: Is this URL provisional?

I get:

`This page doesn’t seem to exist`

Comment (lrknox) [reply README.md:113]: I do, too.  Same result for https://support.hdfgroup.org/archive/support/ftp/HDF5/releases/:  "This page doesn’t seem to exist".

Maybe https://support.hdfgroup.org/downloads/index.html?

**Comment by byrnHDF** [comment]
Latest change should fix the problem

**Comment by mattjala** [comment]
https://support.hdfgroup.org/releases/hdf5/v1.14/index.html Seems to also be a 404

**Comment by byrnHDF** [comment]
Should be an underscore

**Review by mattjala — APPROVED** [review]
Third time's the charm!

File: README.md [hunk 2]
```diff
@@ -103,14 +103,18 @@ in which a feature is introduced may change.
...
 
 Source packages for current and previous releases are located at:
-    
-   https://support.hdfgroup.org/downloads/hdf5
+
+   hdf5 1.14 releases:
+   https://support.hdfgroup.org/releases/hdf5/v1_14/index.html
+
+   Archived releases:
+   https://support.hdfgroup.org/archive/support/ftp/HDF5/releases/
```
Comment (lrknox) [inline README.md]: The other URL works, but this one doesn't

**Review by lrknox — APPROVED** [review]

**Pull request merged**